src/components/select/simple-select/simple-select.stories.mdx

```diff
@@ -127,6 +127,105 @@ You can use the `required` prop to indicate if the field is mandatory.
   </Story>
 </Preview>
 
+### With object as value:
+Option values could be passed as objects. Useful when custom data is associated with an option.
+
+<Preview>
+  <Story name="with object as value" parameters={{ chromatic: { disable: true }}} >
+    {() => {
+      const [value, setValue] = useState({});
+      const optionList = useRef([
+        <Option text='Amber' key='Amber' value={ { value: 1, text: 'Amber' } } />,
+        <Option text='Black' key='Black' value={ { value: 2, text: 'Black' } } />,
+        <Option text='Blue' key='Blue' value={ { value: 3, text: 'Blue' } } />,
+        <Option text='Brown' key='Brown' value={ { value: 4, text: 'Brown' } } />,
+        <Option text='Green' key='Green' value={ { value: 5, text: 'Green' } } />,
+        <Option text='Orange' key='Orange' value={ { value: 6, text: 'Orange' } } />,
+        <Option text='Pink' key='Pink' value={ { value: 7, text: 'Pink' } } />,
+        <Option text='Purple' key='Purple' value={ { value: 8, text: 'Purple' } } />,
+        <Option text='Red' key='Red' value={ { value: 9, text: 'Red' } } />,
+        <Option text='White' key='White' value={ { value: 10, text: 'White' } } />,
+        <Option text='Yellow' key='Yellow' value={ { value: 11, text: 'Yellow' } } />
+      ]);
+      function onChangeHandler(event) {
+        setValue(event.target.value);
+      }
+      function clearValue() {
+        setValue({});
+      }
+      return (
+        <div>
+          <Button onClick={ clearValue } mb={ 2 }>clear</Button>
+          <Select
+            id='with-object'
+            name='with-object'
+            value={ value }
+            onChange={ onChangeHandler }
+          >
+            { optionList.current }
+          </Select>
+        </div>
+      );
+    }}
+  </Story>
+</Preview>
+
+### With isLoading prop:
+When `isLoading` prop is passed, a loader will be appended at the end of the Select List. That functionality could be used to load the options asynchronously.
+
+This prop should only be used when the dropdown is expanded.
```

Comment:
```suggestion
```

Reply:
if the dropdown is not expanded this prop will do nothing